js-envs/src/main/scala/org/scalajs/jsenv/AsyncJSRunner.scala

```diff
@@ -38,12 +38,25 @@ trait AsyncJSRunner {
 
   /** Await completion of the started Run for the duration specified
    *  by [[atMost]]. Strictly equivalent to:
-   * 
+   *
    *  {{{
    *  Await.result(future, atMost)
    *  }}}
-   * 
+   *
    */
   final def await(atMost: Duration): Unit = Await.result(future, atMost)
 
+  /** Await completion of the started Run for the duration specified by
+   *  [[atMost]], or force it to stop if the timeout expires.
+   */
+  final def awaitOrStop(atMost: Duration): Unit = {
+    try {
+      await(atMost)
+    } catch {
+      case th: Throwable =>
```

Comment:
No. If the `future` **terminates** with an exception, you must not call stop, but throw that exception. An example is when the VM terminates with a non-zero exit code.